modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java

```diff
@@ -254,81 +267,66 @@ public void testSseApiThrottling() throws Exception {
+        createAPIRevisionAndDeployUsingRest(apiId, restAPIPublisher);
+        waitForAPIDeploymentSync(user.getUserName(), apiName, apiVersion, APIMIntegrationConstants.IS_API_EXISTS);
+        APIDTO api = restAPIPublisher.getAPIByID(apiId);
+        Assert.assertNotNull(api.getPolicies());
+        Assert.assertTrue(api.getPolicies().contains(policyName));
+
+        // Subscribe to the API using a new application with custom policy as the tier
+        HttpResponse applicationResponse = restAPIStore.createApplication(applicationName + "ThrottleTest",
+                "", APIMIntegrationConstants.API_TIER.UNLIMITED, ApplicationDTO.TokenTypeEnum.OAUTH);
+        throttleAppId = applicationResponse.getData();
+        SubscriptionDTO subscriptionDTO = restAPIStore.subscribeToAPI(apiId, throttleAppId, policyName);
+        // Validate Subscription of the API
+        Assert.assertEquals(subscriptionDTO.getStatus(), SubscriptionDTO.StatusEnum.UNBLOCKED);
+        ArrayList grantTypes = new ArrayList();
+        grantTypes.add(APIMIntegrationConstants.GRANT_TYPE.PASSWORD);
+        grantTypes.add(APIMIntegrationConstants.GRANT_TYPE.REFRESH_CODE);
+        grantTypes.add(APIMIntegrationConstants.GRANT_TYPE.CLIENT_CREDENTIAL);
+        ApplicationKeyDTO applicationKeyDTO = restAPIStore.generateKeys(throttleAppId, "3600", null,
+                ApplicationKeyGenerateRequestDTO.KeyTypeEnum.PRODUCTION, null, grantTypes);
+        String accessToken = applicationKeyDTO.getToken().getAccessToken();
+
+        testThrottling(accessToken);
+        sseServlet.setEventsSent(0);
+        sseReceiver.setReceivedDataEventsCount(0);
     }
 
     private void testThrottling(String accessToken) throws Exception {
-        // Attempt to run for 3 minutes. Connection is expected to be closed within 1 minute due to throttle out.
-        startAndStopSseServer(3 * 60000);
+        sseServer.start();
         // Prevent API requests getting dispersed into two time units
-        while (LocalDateTime.now().getSecond() > 40) {
+        while (LocalDateTime.now().getSecond() > 30) {
             Thread.sleep(5000L);
         }
         long startTime = System.currentTimeMillis();
         startSseReceiver(accessToken, null);
         long endTime = System.currentTimeMillis();
-        Assert.assertTrue((endTime - startTime) < 60000);
-
-        // Re-attempt to connect after throttle out. A throttled out event is expected to be received.
-        AtomicBoolean isThrottled = new AtomicBoolean(false);
-        startSseReceiver(accessToken, new Consumer<Boolean>() {
-            @Override
-            public void accept(Boolean aBoolean) {
-                isThrottled.set(aBoolean);
-            }
-        });
-        Thread.sleep(3000L);
-        Assert.assertTrue(isThrottled.get());
+        long connectionAliveDuration = endTime - startTime;
+        Assert.assertTrue(connectionAliveDuration < 60000);
+        Assert.assertTrue(connectionAliveDuration > 0);
```

Comment:
here too

Reply:
and are we validating throttling based on time, not event counts ?

Reply:
Done via https://github.com/wso2/product-apim/pull/10992/commits/99f4805af2b1b07ed841e5e5bd000c5178af5384